_posts/2021-01-11-howto-nuxeo-package.md

```diff
@@ -0,0 +1,245 @@
+      <mailSessionJndiName>java:comp/env/Mail</mailSessionJndiName>
+    </settings>
+  </extension>
+
+</component>
+$ docker rm -f nuxeo
+```
+
+We can see that our configuration has been taken into account.
+
+
+Specifying Another Email Prefix For The DEV Environment
+=======================================================
+
+[ACME] is the default prefix. But for a development environment, it may be a problem that the prefix is the same than in production: it could lead to misinterpreation on some test messages. To change that configuration in dev, we could ask the devops team to manage that configuration at the infrastructure level and override it in the global `nuxeo.conf`. The other and better way of doing it is to add a `nuxeo.dev` file next to the `nuxeo.defaults` one and fill it with:
+```
+nuxeo.notification.eMailSubjectPrefix="[ACME Dev] "
+```
+
+When starting Nuxeo, setting the `NUXEO_ENVIRONMENT` variable to `dev` will append the content of that file to the configuration. That way, Ops team just have to set the `NUXEO_ENVIRONMENT` variable for the correct configuration to be picked up.
+
+After adding that file, we will test our DEV environment with the same Docker pattern:
+
+
+```shell
+$ mvn clean package
+$ docker build -t nuxeo-acme --build-arg NUXEO_CLID .
+$ docker run -d --name nuxeo nuxeo-acme
+425d138bc6d9b49dee92791296a3a60ec92718dd6b066c9f2821b03e24a6d093
+$ docker exec nuxeo cat /opt/nuxeo/server/nxserver/config/notification-config.xml | grep eMailSubjectPrefix
+      <eMailSubjectPrefix>"[ACME] " </eMailSubjectPrefix>
+$ docker rm -f nuxeo
+nuxeo
+$ docker run -d -e NUXEO_ENVIRONMENT=dev --name nuxeo nuxeo-acme
+$ docker exec nuxeo cat /opt/nuxeo/server/nxserver/config/notification-config.xml | grep eMailSubjectPrefix
+      <eMailSubjectPrefix>"[ACME Dev] " </eMailSubjectPrefix>
+$ docker rm -f nuxeo
+nuxeo
+$
+```
+
+We can see that in the second run, by running the same image, hence the same package, we have been able to have a different configuration specific to the dev environment.
+
+
+A More Sophisticated Example
+============================
+
+In the previous example, we just configured an existing Nuxeo configuration parameter bound to an environment. In Nuxeo, the configuration might
+also take the form of contribution to the extension point of a component. Let's say that we want to change the background color of the login box
+in our dev environment to orange so that it can't be taken for the production one. We need to look at [how to customize the login page](https://doc.nuxeo.com/nxdoc/how-to-customize-the-login-page/) to see that we need to extend the `PluggableAuthenticationService`.
+
+Let's add the following file in `nuxeo-acme-package/src/main/resources/install/templates/marketplace-acme/config/login-screen-config.xml.nxftl`:
+
+```
+<component name="org.nuxeo.login.startup.page.web.contrib.override">
+<#if !((org.nuxeo.prod.environment)??) || (org.nuxeo.prod.environment) != "true">
```

Comment:
I wonder if the property name here should be something different to avoid reader confusion around the `NUXEO_ENVIRONMENT` variable provided to docker, and the property being listed in `nuxeo.prod` that drives this logic.

Maybe if it wasn't about it being prod, and was more about using alternative login color feature flag...  I found myself trying to see how you made the leap from NUXEO_ENVIRONMENT=prod to org.nuxeo.prod.environment where really they are entirely unrelated accept for the setting in `nuxeo.prod`.